### PR TITLE
NetObserv 1.7 FlowCollector API regeneration

### PR DESCRIPTION
**Pull request by skrthomas**
Note to reviewers: This is autogenerated from https://github.com/netobserv/network-observability-operator/pull/796/files, which is still open and could accept some changes but some suggestions may be deferred until the next regeration of the API. 
Version(s):

Merge only to no-1.7. On GA 10/21/24, Sara will open a branch to integrate no-1.7 with main, and that branch will go to 4.12+.
Issue:

This PR spans multiple tickets, as it comprises API updates for the entire release. 
Link to docs preview:

https://83175--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/network_observability/flowcollector-api.html
QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:2770]: see https://github.com/netobserv/network-observability-operator/pull/796/files#r1797373805

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:2803]: This is an annotation, so n/a

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:2807]: n/a bc this is part of the annotation

**Comment by skrthomas** [comment]
@jotak @memodi I just need approvals here for merge. I commented a couple minor things on the regeneration PR. https://github.com/netobserv/network-observability-operator/pull/796/files#

**Comment by jotak** [comment]
Hi @skrthomas [I updated my PR](https://github.com/netobserv/network-observability-operator/pull/796) do you want to check before we merge this one?

**Review by memodi — APPROVED** [review]
/label qe-approved

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:215]:
```suggestion
- `NetworkEvents`: enable the network events monitoring feature, such as correlating flows and network policies.
```
Should "Network" be capitalized here?

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:218]:
```suggestion
IMPORTANT: This feature is available as a Developer Preview. +
```

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:417]: Should line 417 be, "`pktDrops` filters flows with packet drops"?

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:464]:
```suggestion
`NetObservDroppedFlows` is triggered when the eBPF agent is missing packets or flows, such as when the BPF hashmap is busy or full, or the capacity limiter is triggered. +
```
Should "BPF hashmap" be "eBPF hashmap"?

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1290]:
```suggestion
By default, Network Observability format proposal is used: https://github.com/rhobs/observability-data-model/blob/main/network-observability.md#format-proposal.
```

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1291]:
```suggestion
As there is currently no accepted standard for L3 or L4 enriched network logs, you can override it with your own.
```

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1311]:
```suggestion
| Address of the OpenTelemetry receiver.
```
Adding closing punctuation for consistency with other descriptions.

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1315]:
```suggestion
| Port for the OpenTelemetry receiver.
```
Same as above.

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1327]:
```suggestion
By default, Network Observability format proposal is used: https://github.com/rhobs/observability-data-model/blob/main/network-observability.md#format-proposal.
```

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1328]:
```suggestion
As there is currently no accepted standard for L3 or L4 enriched network logs, you can override it with your own.
```

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1412]:
```suggestion
| Specify how often metrics are sent to a collector.
```

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1434]:
```suggestion
| `caCert` defines the reference of the certificate for the Certificate Authority.
```

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1447]:
```suggestion
| `userCert` defines the user certificate reference and is used for mTLS. When you use one-way TLS, you can ignore this property.
```

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1454]:
```suggestion
`caCert` defines the reference of the certificate for the Certificate Authority.
```

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1493]:
```suggestion
`userCert` defines the user certificate reference and is used for mTLS. When you use one-way TLS, you can ignore this property.
```

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:2534]:
```suggestion
It provides flexibility in the network policy configuration, but if you need a more specific
```

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:2541]:
```suggestion
Either enable it, or create your own network policy for Network Observability.
```

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:2703]:
```suggestion
To guarantee a correct identification, the indexed values must form an unique identifier across the cluster.
```

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:2704]:
```suggestion
If the same index is used by several resources, those resources might be wrongly labeled.
```

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:2776]:
```suggestion
To guarantee a correct identification, the indexed values must form an unique identifier across the cluster.
```

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:2777]:
```suggestion
If the same index is used by several resources, those resources might be wrongly labeled.
```

**Review by lahinson — COMMENTED** [review]
Just a few suggestions for your consideration!

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:215]: No it shouldn't

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:218]: I also think that `+` shouldn't be there. Good point though about this one.

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:417]: I think you're right here.

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:125]: In my file, this comma precedes a continuation of the sentence: 

> `processor` defines the settings of the component that receives the flows from the agent,
_enriches them, generates metrics, and forwards them to the Loki persistence layer and/or any available exporter._

Comment (lahinson) [reply modules/network-observability-flowcollector-api-specifications.adoc:125]: Oh! That's cut off for me here. I should have expanded the file. Feel free to ignore my comment.

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:464]: Yeah it should be eBPF I think.

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:1290]: I think there's a space here so the period doesn't interfere with the link

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:2704]: I also think incorrectly is better than wrongly. Thanks for pointing this out.

**Comment by skrthomas** [comment]
Thanks for the review @lahinson !

File: modules/network-observability-flowcollector-api-specifications.adoc
Comment (jotak) [inline modules/network-observability-flowcollector-api-specifications.adoc:205]: hey @skrthomas , as commented in my PR, this change is incorrect

**Pull request merged**
